python/ray/dag/tests/experimental/test_torch_tensor_dag.py

```diff
@@ -1385,31 +1394,40 @@ def test_torch_tensor_nccl_comm_deduplicate_collective_and_p2p(ray_start_regular
         ]
         collectives = collective.allreduce.bind(computes)
         recvs = [
-            workers[(i + 1) % num_workers].recv.bind(
-                collective.with_type_hint(TorchTensorType(transport="nccl"))
-            )
-            for i, collective in enumerate(collectives)
+            # Each of the 2 worker recvs from the other.
+            workers[0].recv.bind(
+                collectives[1].with_type_hint(TorchTensorType(transport="nccl"))
+            ),
+            workers[1].recv.bind(
+                collectives[0].with_type_hint(TorchTensorType(transport="nccl"))
+            ),
         ]
         dag = MultiOutputNode(recvs)
 
     compiled_dag = dag.experimental_compile()
 
     from ray.dag.collective_node import CollectiveOutputNode
+    from ray.experimental.channel import ChannelContext
 
-    comms = set()
+    comm_ids = set()
     for _, exec_tasks in compiled_dag.actor_to_executable_tasks.items():
         for exec_task in exec_tasks:
             dag_node = compiled_dag.idx_to_task[exec_task.task_idx].dag_node
             if isinstance(dag_node, CollectiveOutputNode):
                 assert exec_task.collective_group is not None
                 nccl_group_id = exec_task.collective_group._type_hint._nccl_group_id
                 assert nccl_group_id is not None
-                comms.add(nccl_group_id)
+                comm_ids.add(nccl_group_id)
     # Only 1 NCCL group should be created.
-    assert len(comms) == 1
-    comm = list(comms)[0]
+    assert len(comm_ids) == 1
+    comm_id = list(comm_ids)[0]
+    ctx = ChannelContext.get_current()
+    comm = ctx.nccl_groups[comm_id]
+    comm_actors = comm.get_actor_handles()
+    # The NCCL group should contain both workers.
+    assert set(comm_actors) == set(workers)
     # The comm for all-reduce should be the same as the p2p send/recv comm.
```

Comment:
Change all "comm" to "nccl_group". Change comments also.

Reply:
done